src/hooks/chart.ts

```diff
@@ -809,463 +795,621 @@ const parseViolinResponse = (
+      // if there is a legend_markdown_pattern then create the link and add it to the array
+      const linkPattern = Array.isArray(legend_markdown_pattern)
+        ? legend_markdown_pattern[index]
+        : legend_markdown_pattern;
+      const extractedLinkPattern = createLinkWithContextParams(linkPattern);
+      if (extractedLinkPattern) {
+        const link = createLink(extractedLinkPattern, { $self: { data: item }, $row: item });
+        if (link) {
+          result.legend_clickable_links.push(link);
+        }
+      }
+    }
 
-  return result;
-};
+    const graphic_link_pattern = trace.graphic_link_pattern || extraInfo?.graphic_link_pattern; // use either the trace or extraInfo
+    if (graphic_link_pattern) {
+      // if there is a graphic_link_pattern then create the link and it to the array
+      const linkPattern = Array.isArray(graphic_link_pattern)
+        ? graphic_link_pattern[0]
+        : graphic_link_pattern;
+      if (linkPattern) {
+        const link = createLink(linkPattern, { $self: { data: item }, $row: item });
+        if (link) {
+          result.graphic_clickable_links.push(link);
+        }
+      }
+    }
 
-/**
- * Parses response data obtained for every trace within the plot
- *
- * @param trace current trace the plot config
- * @param plot plot config
- * @param responseData data received from request to be parsed
- * @returns
- */
-const parsePieResponse = (trace: Trace, plot: Plot, responseData: ResponseData) => {
-  const result: Partial<TraceConfig> & Partial<PlotlyPieData> & PieResultData & ClickableLinks = {
-    ...trace,
-    type: plot.plot_type,
-    hoverinfo: 'text+value+percent', // value to show on hover of a pie slice
-    textinfo: plot.config.slice_label || 'value', // data shown on a pie slice
-    labels: [], // array of labels for the legend
-    text: [], // text to show on hover of a pie slice
-    values: [], // the values for the legend
-    legend_clickable_links: [], // array of links for when clicking legend
-    graphic_clickable_links: [], // array of links for when clicking graph
   };
 
-  const { config } = plot;
-  const { format_data = false } = config;
-  responseData.forEach((item: any, i: number) => {
-    updateWithTraceColData(result, trace, item, i);
-    // Add data
-    if (trace.data_col) {
-      const value = getValue(item, trace.data_col, undefined, format_data, plot);
-      if (Array.isArray(result.values)) {
-        result.values.push(value);
+  /**
+   * 
+   * @param result data object to be updated
+   * @param trace current trace from the plot config
+   * @param item each item/row of data
+   * @param textArray array for hover text
+   * @param index provided for heatmap case for corresponding y axis values
+   * @param templateParams provided for heatmap to access url_parameters
+   * @returns 
+   */
+  const updateHoverTemplateData = (
+    result: any,
+    trace: Trace,
+    item: any,
+    textArray: string[] & string[][],
```

Comment:
Should this be `string[] & string[][]` or `string[] | string[][]`?

I'm not sure what the difference is if you could comment why one is preferred over the other. 

I think the case that might be `string[][]` is for heatmap plot type since the `text` array matches the `z` array?